src/Twig/Extension/LanguageSwitcherExtension.php

```diff
@@ -30,12 +32,18 @@ class LanguageSwitcherExtension extends AbstractExtension
      */
     private $documentService;
 
-    public function __construct(Service $documentService)
+    /**
+     * @var UrlGeneratorInterface $urlGenerator
+     */
+    private UrlGeneratorInterface $urlGenerator;
+
```

Comment:
```suggestion
   /**
    * @var RequestStack
    */
    private $requestStack;
```

Reply:
@yuliusardian We can get the current request from request stack in the construct like https://github.com/pimcore/demo/blob/d0561bd6a7e373ae2d90d774e68286499a5c43bf/src/Website/LinkGenerator/AbstractProductLinkGenerator.php#L50

Reply:
Sorry for late response, I've been busy...

Sure, I knew that..